### ISSUE TITLE
Bug: Cannot Remove Contact Once Ingested By Signal

### ISSUE THREAD

**Issue by Enekk** [signalapp/Signal-Android#6014]
I have:
- [x] searched open and closed issues for duplicates
- [x] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

----------------------------------------

### Bug description
Once a contact phone number has been ingested by Signal (from Android's address book), it cannot be removed even if it is removed from Android itself.
This became an issue when a typo inadvertently connected me with another Signal user and I removed the incorrect phone number from my address book.
The only way to actually remove the contact was to, after removing it from Android's address book, completely wipe and reinstall Signal; luckily it was early enough in my usage of Signal that this wasn't an issue.

### Steps to reproduce
- Add a phone number to Android's address book
- Contact that user
- Remove the contact from Android's address book

**Actual result:** The contact remains in Signal
**Expected result:** I would expect the contact to remain in Signal, but I would also expect a "forget user" option somewhere.
This would also be useful if the user was compromised or changed phone numbers.

### Screenshots
None to include.

### Device info
**Device:** Nexus 6p
**Android version:** 7.1.1
**Signal version:** 3.25.4

### Link to debug log
None provided

**Comment by riyapenn** [comment]
Resync your contacts
1. Go to Android Settings
2. Choose Accounts
3. Choose Signal
4. Tap on the Menu
5. Choose Remove Account (the alert of clearing data is incorrect, your messages will not be deleted)
6. Open Signal
7. Tap on the pencil icon in the blue circle
8. Choose the Menu
9. Tap Refresh

**Comment by Enekk** [comment]
Well, I'm glad there is a way to do this, but that seems much more complicated than necessary.  Thanks for the information anyway!

**Comment by moxie0** [comment]
What do you mean by "the contact remains in signal?"

**Comment by riyapenn** [comment]
Consideration: Trigger an update to all linked devices

**Comment by Enekk** [comment]
@moxie0: By the "Contact Remains in Signal", I mean:
1. I removed the contact from Android's address book
2. I long pressed on the conversation with the bad contact and deleted it

At this state I was still able to see the bad contact when I hit new message button (the pencil).  If I then clicked on the bad contact's picture, I was taken to the Android Address Book which showed the correct data except that there was a set of contact information specifically for signal that had the wrong information in it.  I was unable to remove the bad contact from here as the Signal specific contact pieces do not become editable when you use Android's 'edit' function.  

I'd love to take some screenshots, but I think Signal prevents this (great feature!).

**Comment by p4nci** [comment]
@Enekk you can enable screenshots (temporary):
Settings -> Privacy -> Screen security

**Comment by NetScr1be** [comment]
I have a related issue. Signal surprises me saying a number in my contacts is on Signal and assigns a name to it. The new contact and I have an awkward conversation to figure out they are not who I think they are.

What happened was the number is in use by a new person but it is in my contacts from years ago.

I updated the Android address book but Signal still thinks it belongs to the wrong person.

There is nowhere to turn off the groovy address book autosearch feature. This has happened multiple times. My Android address list is old and extensive.

**Comment by Enekk** [comment]
@NetScr1be This is exactly how I got the point I was at as well.

**Comment by bellja** [comment]
I've this happen to me. 

I got a notification that someone in my google contacts "has joined Signal", initially i was excited. I hadn't talked to them in a while, so after messaging briefly we realised my contact was not the person i thought they were. My contact had apparently changed phone numbers, so I didn't really know who i was talking to at this point. 

I deleted the phone number from the entry in my google contacts app, and tried to resync with Signal. No effect. Signal still thinks this phone number (with Name) is a valid option for a new conversation. Technically true, if i wanted to chat with a stranger. 

**There need to be an option to force an update of the internal Signal contact list.**
